rust/src/rfb/rfb.rs

```diff
@@ -686,43 +719,43 @@ mod test {
 
         //The buffer values correspond to Server Protocol version: 003.008
         // Same buffer is used for both functions due to similar values in request and response
-        init_state.parse_response(&buf[0..12]);
+        init_state.parse_response(std::ptr::null(), StreamSlice::from_slice(&buf[0..12], STREAM_START, 0));
         let mut ok_state = parser::RFBGlobalState::TSClientProtocolVersion;
         assert_eq!(init_state.state, ok_state);
 
         //The buffer values correspond to Client Protocol version: 003.008
-        init_state.parse_request(&buf[0..12]);
+        init_state.parse_request(std::ptr::null(), StreamSlice::from_slice(&buf[0..12], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TCSupportedSecurityTypes;
         assert_eq!(init_state.state, ok_state);
 
-        init_state.parse_response(&buf[12..14]);
+        init_state.parse_response(std::ptr::null(), StreamSlice::from_slice(&buf[12..14], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TSSecurityTypeSelection;
         assert_eq!(init_state.state, ok_state);
 
-        init_state.parse_request(&buf[14..15]);
+        init_state.parse_request(std::ptr::null(), StreamSlice::from_slice(&buf[14..15], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TCVncChallenge;
         assert_eq!(init_state.state, ok_state);
 
         //The buffer values correspond to Server Authentication challenge: 547b7a6f36a154db03a2575c6f2a4ec5
         // Same buffer is used for both functions due to similar values in request and response
-        init_state.parse_response(&buf[15..31]);
+        init_state.parse_response(std::ptr::null(), StreamSlice::from_slice(&buf[15..31], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TSVncResponse;
         assert_eq!(init_state.state, ok_state);
 
         //The buffer values correspond to Client Authentication response: 547b7a6f36a154db03a2575c6f2a4ec5
-        init_state.parse_request(&buf[15..31]);
+        init_state.parse_request(std::ptr::null(), StreamSlice::from_slice(&buf[15..31], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TCSecurityResult;
         assert_eq!(init_state.state, ok_state);
 
-        init_state.parse_response(&buf[31..35]);
+        init_state.parse_response(std::ptr::null(), StreamSlice::from_slice(&buf[31..35], STREAM_START, 0));
         ok_state = parser::RFBGlobalState::TSClientInit;
         assert_eq!(init_state.state, ok_state);
 
-        init_state.parse_request(&buf[35..36]);
+        init_state.parse_request(std::ptr::null(), StreamSlice::from_slice(&buf[35..36], STREAM_START, 0));
```

Comment:
I'm still unable to see how this fn sig is reflecting frames.. 🤔 
Why can't we change the fn signature as one commit and then add frames in a later one?
Could you please elaborate what problems are you running into if you do that?

Reply:
yes. Because i don't see that the signature has anything to do with the frames even when you change it for the frames. The signature makes you take a `StreamSlice` instead of a byte array and an additional param which you could have added anyway and wouldn't make a difference in the code and execution. 😉

Reply:
this got merged. Shoudl I still be making a new PR with this change?

Reply:
I think that is a nice cleanup, so yeah please have a look at it.

Reply:
Since the code got merged. I have a little confused about how the commits will be organized now. If I rebase my code and try to separate this into two commits how would that be now?

Reply:
You take the `master` branch, create your own new branch for it, make your new changes.

Reply:
Alright thanks